greenpeace/planet4-master-theme
# PoC: Linting on pre-commit

- lint-staged to run linting on staged files
- script to wrap wp-env calls if available
- script to get around phpcbf exit code issue

![Screenshot from 2023-08-23 19-45-22](https://github.com/greenpeace/planet4-master-theme/assets/617346/9a73eb77-7725-4610-834a-a0e09e10129f)

## Test

Run `npm install` first, verify that `git config core.hooksPath` returns `.githooks`.

- run lint-staged by itself with `npx lint-staged`
  - edit a PHP file (remove a `,` from `src/ActionPage.php l.30`
  - lint should run and fix some issues automatically
- edit some js file, run `git add && git commit`
  - lint should run and fix some issues automatically
